src/others/gridnav/lv_gridnav.c

```diff
@@ -85,9 +85,11 @@ void lv_gridnav_remove(lv_obj_t * obj)
     }
 
     if(event_dsc) {
-        lv_free(lv_event_dsc_get_user_data(event_dsc));
-        lv_obj_remove_event(obj, i);
+        if(lv_event_dsc_get_cb(event_dsc) == gridnav_event_cb) {
```

Comment:
I think it’s better to move the code inside the loop if statement, above the `break`, it’s simpler and saves us one more comparison.

Reply:
I agree.